gui/web/src/utils/metricsTracker.js

```diff
@@ -0,0 +1,27 @@
+class MetricsTracker {
+    constructor(apiKey) {
+        this.apiKey = apiKey;
+        this.apiUrl = "https://api2.amplitude.com/2/httpapi";
+
+        // TODO: Implement more functions.
+        this.sendEvent = this.sendEvent.bind(this);
+    }
+
+    sendEvent(event) {
+        var _this = this
+        this._asyncRequests["fetch"] = fetch(this.apiUrl, {
```

Comment:
we shouldn't use the `fetch` function directly. everything should be in the kelp-ops-api folder (see version.js as an example).

this structures the code well when trying to streamline and track where we make all our network requests.

Reply:
we also don't want all requests to use the tag `fetch` since we don't want the sending of two different requests to cancel out each other. we can call it `"sendEvent/" + event` so each event gets its own namespace.

does the purpose of _asyncRequests make sense? (we use this pattern quite heavily across the app)

Reply:
Agreed that we shouldn't use the `fetch` function directly, but do we want to use `kelp-ops-api` for external requests? All existing endpoints in `kelp-ops-api` are for routes for the Kelp API (i.e., `api/v1/stop` or `api/v1/optionsMetadata`). Mixing internal and external use cases seems like a fast track to confusing directory structure. Let me know if there are already external use cases in that structure and I just missed it; if not, might be good to consider best organization. Thoughts on the right way to proceed here?

And yes, the purpose of `_asyncRequests` makes total sense. If I'm understanding the namespacing correctly, I can definitely change the namespace to `"sendEvent/" + event` - by modifying the key in `this._asyncRequests`, yeah?

Reply:
yeah, I agree -- we can create a new folder called `kelp-external-api` and put it in there. although with the above comment that will be routed through an internal request now. but good call!

re namespace: yes, we need to modify the key on the _asyncRequests map and that's pretty much it